app/views/partials/_counter.html.erb

```diff
@@ -1,23 +1,31 @@
 <div class="count">
-  <span>Comments:
-      <% if post.comments_counter %>
-      <%= post.comments_counter %>
-      <% else %>
-      0
-      <% end %>
-  </span>
+    <span>Comments:
+        <% if post.comments_counter %>
+        <%= post.comments_counter %>
+        <% else %>
+        0
+        <% end %>
+    </span>
 
-  <span>Likes:
-      <% if post.likes_counter %>
-      <%= post.likes_counter %>
-      <% else %>
-      0
-      <% end %>
-  </span>
-  
-  <%= form_with(model: @like, url: user_post_likes_path(post.author, post)) do |form| %>
-      <%= form.button type: :submit do %>
-          <span class="thumb"> &#128077; </span>
-      <% end %>
-  <% end %>
+    <span>Likes:
+        <% if post.likes_counter %>
+        <%= post.likes_counter %>
+        <% else %>
+        0
+        <% end %>
+    </span>
+    
+    <%= form_with(model: @like, url: user_post_likes_path(post.author, post)) do |form| %>
+        <%= form.button type: :submit do %>
+            <span class="thumb"> &#128077; </span>
+        <% end %>
+    <% end %>
+
+    <% if can? :destroy, post %>
+        <div class="btn-delete">
+            <%= link_to user_post_url(post.author_id, post), method: :delete, class: 'btn-link' do %>
+                <button>Delete Post</button>
+            <% end %>
+        </div>
+    <% end %>
```

Comment:
- [ ] I noticed the delete operation for a post is not functioning as it should. Kindly work on it so a post can be deleted as expected 👍🏽

Reply:
Fixed.